libc/src/ckb_cell_fs.c

```diff
@@ -41,7 +50,37 @@ int get_file(const CellFileSystem *fs, const char *filename, FSFile **f) {
 
 int ckb_get_file(const char *filename, FSFile **file) { return get_file(CELL_FILE_SYSTEM, filename, file); }
 
-int load_fs(CellFileSystem **fs, void *buf, uint64_t buflen) {
+/*
+prefix_normalize("") == "/"
+prefix_normalize("/") == "/"
+prefix_normalize("foo") == "/foo/"
+prefix_normalize("foo/") == "/foo/"
+prefix_normalize("/foo") == "/foo/"
+prefix_normalize("/foo/bar") == "/foo/bar/"
+prefix_normalize("/foo/bar/") == "/foo/bar/"
+prefix_normalize("foo/bar") == "/foo/bar/"
+prefix_normalize("foo/bar/") == "/foo/bar/"
+*/
+char *prefix_normalize(const char *prefix) {
+    int len = strlen(prefix);
+    if (len == 0) {
+        return "/";
+    }
+    char *buf = malloc(len + 3);
```

Comment:
When to free this `buf`? actually only checking is required. The normalizing is not necessary.